### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by RCasatta**
~~On top of #13 see only cf81501~~

Adds an integration test using bitcoind regtest as backend for payjoin receiver and sender using the implementation provided by the lib itself (the receiver part isn't finished since the library hasn't that part yet)

Note it uses my crate bitcoind which is a helper to spawn bitcoind processes

### CODE REVIEW

**Comment by Kixunil** [comment]
Yeah, I was thinking about something like that. I wonder if we can get rid of needing bitcoind for internal test to make it faster. I think it should be possible (using `bitcoin` crate only).

I also intend to make end-to-end test using [lnpbp-testkit](https://github.com/Kixunil/lnpbp-testkit) and multiple nodes (that part is not yet supportd in the testkit but planned)

**Comment by RCasatta** [comment]
Not sure to follow you, you don't want bitcoind for faster tests but do you want to make end-to-end test using lnpbp-testkit and multiple nodes?

Anyway, at the moment the integration test takes 5 seconds, consider also that every other integration test could run in parallel spawning its own node.
If it's considered too much, we can have a specific feature to run tests requiring spawning other processes (like btcpay, bitcoind, lnd, c-lightning or others).
I would probably rather not have a specific feature at the moment, when developing you can easily filter and launch only a specific test or specific test module.

**Comment by Kixunil** [comment]
My thinking is to run end-to-end tests less often (e.g. in CI but not locally during development).

> the integration test takes 5 seconds

Wow, that's impressive! I thought it'd take much longer to download bitcoind binary, launch it, compile the code and do all other stuff. What kind of black magic is that?

> I would probably rather not have a specific feature at the moment

Yep, that's fine.

**Comment by RCasatta** [comment]
> Wow, that's impressive! I thought it'd take much longer to download bitcoind binary, launch it, compile the code and do all other stuff. What kind of black magic is that?

I am not considering compile time and referring to the second launch, the binary is cached in cargo home and not downloaded again if present

**Comment by RCasatta** [comment]
I am not sure about 5071737789fa9d7826ffa00c4dc4d3e2f5776326, I did it because it looks `bitcoind` MSRV is greater than 1.41.1, I will check there if it's feasible to have bitcoind running with `1.41.1` so we don't need 5071737789fa9d7826ffa00c4dc4d3e2f5776326

**Comment by RCasatta** [comment]
> I am not sure about [5071737](https://github.com/Kixunil/payjoin/commit/5071737789fa9d7826ffa00c4dc4d3e2f5776326), I did it because it looks `bitcoind` MSRV is greater than 1.41.1, I will check there if it's feasible to have bitcoind running with `1.41.1` so we don't need [5071737](https://github.com/Kixunil/payjoin/commit/5071737789fa9d7826ffa00c4dc4d3e2f5776326)

yes, i reduced bitcoind MSRV with https://github.com/RCasatta/bitcoind/pull/25, will revisit this one once I release bitcoind

File: payjoin-client/Cargo.toml [hunk 1]
```diff
@@ -8,6 +8,6 @@ edition = "2018"
 
 [dependencies]
 bip78 = { path = "../bip78", features = ["sender"] }
-bitcoincore-rpc = "0.13.0"
+core-rpc = "0.14.0"
```
Comment (Kixunil) [inline payjoin-client/Cargo.toml:11]: What's different in `core-rpc`?

**Review by Kixunil — APPROVED** [review]
Looks good. I will need to take time to learn implications of using `core-rpc` before merging.

File: payjoin-client/Cargo.toml (hunk 1 above)
Comment (RCasatta) [reply payjoin-client/Cargo.toml:11]: Rust-bitcoin version 0.27 in core-rpc

**Pull request merged**